### DISCUSSION TITLE
Welcome to agents-aea Discussions!

### DISCUSSION THREAD

**Discussion by DavidMinarsch** [fetchai/agents-aea#2453] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by amendoz3** [comment]
Hi, I am building this project but I am stuck building the autonomous Economic Agent AEA. When I try to download the packages and scripts
 
svn export https://github.com/fetchai/agents-aea.git/trunk/packages
 
I get the following error
 
amendoz3@raspberrypi:~ $ svn export https://github.com/fetchai/agents-aea.git/trunk/packages
bash: svn: command not found
 
I am using raspberry pi 3.
 
What can I do to fix this error? and biuld AEA

**Comment by amendoz3** [comment]
Also, I am using a mac with MacOS Big Sur Version 11.7

**Reply by VijaySharma815** [reply]
`brew install svn`

You can check the version of svn: `svn --version`

**Comment by amendoz3** [comment]
When I enter the command
svn –version

I get the following:
***@***.***:~ $ svn--version
bash: svn--version: command not found

From: VijaySharma815 ***@***.***>
Date: Wednesday, October 5, 2022 at 3:35 AM
To: fetchai/agents-aea ***@***.***>
Cc: Mendoza Alvarez, Alexis (amendoz3) ***@***.***>, Mention ***@***.***>
Subject: Re: [fetchai/agents-aea] Welcome to agents-aea Discussions! (Discussion #2453)

Caution! This E-mail originated from outside the university. Please use caution when opening attachments, clicking links, or responding to requests.



@amendoz3<https://github.com/amendoz3> , could you please check the svn version: svn --version ?

**Comment by amendoz3** [comment]
<img width="328" alt="Screen Shot 2022-10-05 at 9 50 02 AM" src="https://user-images.githubusercontent.com/115052250/194091042-80839fd0-e499-4180-904c-99209e2e6dfc.png">

Here is what I am getting.

**Comment by amendoz3** [comment]
I just installed the svn with the following command 
$ sudo apt-get update 
$ sudo apt-get upgrade
$ sudo apt-get install subversion
 
I was able to succesfully import the scripts and packages. Thank you.

**Reply by VijaySharma815** [reply]
yea! these command works in Linux OS.

**Comment by amendoz3** [comment]
Now I am running into this issue when biulding the agent, 

amendoz3@raspberrypi:~/carpark_aea $ aea add skill carpark_detection
error: Cannot find skill: 'carpark_detection'.

**Reply by VijaySharma815** [reply]
@amendoz3 , can you please share your `aea-config.yaml` file?

**Reply by jrriehl** [reply]
Try `aea add skill fetchai/carpark_detection`

**Comment by sc-arpjos** [comment]
I have few questions:
is it possible to trigger one skill from another skill?
how to create contract between 2 agents based on their services and execute it when certain conditions are met?

**Reply by VijaySharma815** [reply]
@sc-arpjos ,

Technically yes, two or more skills can interact with each other.  In order to do so, we can use shared_state to share an object b/w different skills. So we can create an object with key-value pairs containing different statuses and we can take reference of that object in different skills and update that accordingly in another skill.

For example, You can take a look at this link https://github.com/fetchai/agents-aea/blob/7988aafa74211ef887e26754bba967e11930a60c/packages/fetchai/skills/simple_service_search/handlers.py where at line 97 in handle_search function we are storing the search result in the shared state, and if we want to use this result in different skill, I can access this result using `self.context.shared_state.get(strategy.shared_storage_key, None]` and do your thing accordingly.

**Reply by sc-arpjos** [reply]
Hi @VijaySharma815,

Thanks for the reply.
One more question -> Can we trigger a skill manually!
Currently skills are getting triggered when we launch agents, is there any way to trigger those skills whenever we want!

**Reply by VijaySharma815** [reply]
yes, either by adding the condition which you want to meet for your code to be called inside the act method of the skill behaviour or by adding:
- a connection to the agent (such as http_server) where they are listening for messages and
- a handler for the message that triggers the desired skill's action

**Reply by sc-arpjos** [reply]
Thanks Vijay

**Reply by VijaySharma815** [reply]
- In skill 1, whenever it is time for skill 2 to wake up and get active, put some key in the shared_state with the value True.
- In skill 2, in a ticker or cyclic behaviour, try to read that key in the shared_state. If not found or False do nothing. If True, proceed with the logic.

**Reply by sc-arpjos** [reply]
its working bt not the way i want.
there are 2 skills -> skill 1 and skill 2
when i set bool in skill1 and use IF statement in skill 2 its working but when i set bool in skill2 and use IF statement in skill1 its not working. 
I thought all skills are executing at once bt it seems there is some order in skill execution

**Reply by VijaySharma815** [reply]
skills run in alphabetical order (friction of seconds gap between skills)

**Reply by sc-arpjos** [reply]
yeah it seems that way. Thanks 
can i use a different protocol instead of acn for communication!

**Reply by VijaySharma815** [reply]
ACN is the only way to communicate with Agents. 

Let me know, if you are facing any issues I will try to help you.

**Comment by amendoz3** [comment]
Here it is,

aea_version: 0.1.13
agent_name: carpark_aea
authors: ''
connections:
- oef
default_connection: oef
description: ''
ledger_apis:
- ledger_api:
    addr: alpha.fetch-ai.com
    ledger: fetchai
    port: 80
license: ''
logging_config:
  disable_existing_loggers: false
  version: 1
private_key_paths:
- private_key_path:
    ledger: default
    path: default_private_key.pem
- private_key_path:
    ledger: fetchai
    path: fet_private_key.txt
- private_key_path:
    ledger: ethereum
    path: eth_private_key.txt
protocols:
- default
registry_path: ../packages
skills:
- error
url: ''
version: v1

From: VijaySharma815 ***@***.***>
Date: Thursday, October 6, 2022 at 1:50 AM
To: fetchai/agents-aea ***@***.***>
Cc: Mendoza Alvarez, Alexis (amendoz3) ***@***.***>, Mention ***@***.***>
Subject: Re: [fetchai/agents-aea] Welcome to agents-aea Discussions! (Discussion #2453)

Caution! This E-mail originated from outside the university. Please use caution when opening attachments, clicking links, or responding to requests.



@amendoz3<https://github.com/amendoz3> , can you please share your aea-config.yaml file?

**Reply by VijaySharma815** [reply]
run the command: 
`aea add skill fetchai/carpark_detection:0.27.2`

Ref: https://aea-registry.fetch.ai/details/skill/fetchai/carpark_detection/0.27.2

**Comment by shepherd6844** [comment]
I am wondering through every link I come across.  My journey started with an FET coin sent to the wrong Fetch.ai address and now I am seeking someone at Fetch.ai or anyone who can help me retrieve my coins.  The ERC0-20 "promise" that I came across makes this seem possible even though I don't know who has access to the address.  The funds tied to this are critical for our family and it was my son who made the error.  Can someone point me in the right direction please.
![Screenshot 2025-04-25 190038](https://github.com/user-attachments/assets/2ae003be-8dd4-451d-95fd-512a9679ec08)
